# Carregamento dos dados

O primeiro passo é importar os pacotes que serão utilizado e em seguida carregar os dados

In [1]:
# importando os pacotes necessários
import pandas as pd

In [2]:
# importando a planilha com os dados
planilha_completa = pd.read_excel("CORRETA - analise d edados.xlsx",None)

In [3]:
# verificando as abas do arquivo carregado
planilha_completa.keys()

dict_keys(['Vendas', 'Custo', 'Saida', 'Caixa'])

Com os dados já carregados, irei dividi-lós de acordo com as abas da planilha

In [4]:
# separando cada uma das abas
vendas = planilha_completa['Vendas']
custo = planilha_completa['Custo']
saida = planilha_completa['Saida']
caixa = planilha_completa['Caixa']

# Limpeza e transformação

Agora que essa separação já foi feita posso começar a verificar possiveis inconsistências nos dados e pontos de ajustes para melhor performance. Começando pela tabela 'vendas'

## Vendas

In [5]:
# conferindo as informações de cada coluna da tabela de vendas

vendas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1866 entries, 0 to 1865
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype 
---  ------           --------------  ----- 
 0   ID               1866 non-null   int64 
 1   DataVenda        1866 non-null   object
 2   ModeloComercial  1866 non-null   object
 3   NotaEntrada      1866 non-null   int64 
 4   NotaSaida        1866 non-null   int64 
 5   Vendedor         1866 non-null   object
 6   Cliente          1866 non-null   object
dtypes: int64(3), object(4)
memory usage: 102.2+ KB


Não foram encontrados dados nulos ou faltantes nas colunas, mas a coluna 'DataVenda' por se tratar de uma coluna de datas deve ser formatada como tal

In [6]:
# verificando como a coluna 'DataVenda' está formatada
vendas['DataVenda']

0         02 de Janeiro de 2024
1       05 de fevereiro de 2024
2          3 de Janeiro de 2024
3        6 de fevereiro de 2024
4          4 de Janeiro de 2024
                 ...           
1861    28 de fevereiro de 2024
1862        17 de junho de 2024
1863        13 de junho de 2023
1864     26 de dezembro de 2023
1865    16 de fevereiro de 2024
Name: DataVenda, Length: 1866, dtype: object

In [7]:
# criando listas para passar os nomes dos meses de português para inglês para conseguir fazer a mudança de tipo de dado
""" Essa função também reconhece outras linguas e faz a mudança a depender da definição de localização 
utilizada, mas nesse caso a minha configuração estava com o inlgês como base, e para evitar essas alterações eu 
preferi trocar os nomes dos meses para inglês """

meses_pt = ['de ','janeiro', 'fevereiro', 'março', 'abril', 'maio', 'junho', 'julho', 'agosto', 'setembro', 'outubro', 'novembro', 'dezembro']
meses_en = ['','january', 'february', 'march', 'april', 'may', 'june', 'july', 'august', 'september', 'october', 'november', 'december']

# trocando os nomes dos meses de português para inglês
for en, pt in zip(meses_en, meses_pt):
    vendas['DataVenda'] = vendas['DataVenda'].str.lower().str.replace(pt, en)

# alterando o tipo da coluna para datetime
vendas['DataVenda'] = pd.to_datetime(vendas['DataVenda'], format='%d %B %Y',dayfirst=True)

In [8]:
# verificando como ficaram os dados depois da alteração
vendas.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1866 entries, 0 to 1865
Data columns (total 7 columns):
 #   Column           Non-Null Count  Dtype         
---  ------           --------------  -----         
 0   ID               1866 non-null   int64         
 1   DataVenda        1866 non-null   datetime64[ns]
 2   ModeloComercial  1866 non-null   object        
 3   NotaEntrada      1866 non-null   int64         
 4   NotaSaida        1866 non-null   int64         
 5   Vendedor         1866 non-null   object        
 6   Cliente          1866 non-null   object        
dtypes: datetime64[ns](1), int64(3), object(3)
memory usage: 102.2+ KB


Agora que os tipos dos dados de todas as colunas já estão corretos, irei olhar os valores de cada coluna e como se distribuem

In [9]:
# demonstração dos dados de vendas
vendas.head()

ID  DataVenda ModeloComercial  NotaEntrada  NotaSaida Vendedor  \
0   1 2024-01-02             FMX       448855     885522    Vitor   
1   2 2024-02-05              FH       448856     885523  Jessica   
2   3 2024-01-03              VM       448857     885524  Jessica   
3   4 2024-02-06              FH       448858     885525   Rafael   
4   5 2024-01-04              FH       448859     885526   Rafael   

         Cliente  
0  EficientTrans  
1      MoviFrete  
2    SuperCargas  
3    ExpressoNet  
4      GlobalLog

In [10]:
# verificando os valores únicos de cada coluna
vendas.nunique()

ID                 1866
DataVenda           536
ModeloComercial       4
NotaEntrada        1866
NotaSaida          1866
Vendedor              4
Cliente              13
dtype: int64

In [11]:
# olhando as medidas de resumo das colunas de tipo numérico
vendas.describe()

ID    NotaEntrada      NotaSaida
count  1866.000000    1866.000000    1866.000000
mean    933.500000  449841.049303  886454.500000
std     538.812119   16464.721464     538.812119
min       1.000000       0.000000  885522.000000
25%     467.250000  449320.250000  885988.250000
50%     933.500000  449786.500000  886454.500000
75%    1399.750000  450253.750000  886920.750000
max    1866.000000  999999.000000  887387.000000

In [12]:
# verificando os primeiros valores ordenados por 'NotaEntrada'
vendas.sort_values('NotaEntrada').head()

ID  DataVenda ModeloComercial  NotaEntrada  NotaSaida Vendedor  \
955  956 2024-01-12              VM            0     886477    Andre   
0      1 2024-01-02             FMX       448855     885522    Vitor   
1      2 2024-02-05              FH       448856     885523  Jessica   
2      3 2024-01-03              VM       448857     885524  Jessica   
3      4 2024-02-06              FH       448858     885525   Rafael   

           Cliente  
955      RápidoLog  
0    EficientTrans  
1        MoviFrete  
2      SuperCargas  
3      ExpressoNet

In [13]:
# verificando os últimos valores ordenados por 'NotaEntrada'
vendas.sort_values('NotaEntrada').tail()

ID  DataVenda ModeloComercial  NotaEntrada  NotaSaida Vendedor  \
1862  1863 2024-06-17              VM       450717     887384    Andre   
1863  1864 2023-06-13             FMX       450718     887385    Vitor   
1864  1865 2023-12-26              FH       450719     887386    Andre   
1865  1866 2024-02-16             FMX       450720     887387   Rafael   
1411  1412 2023-05-22             FMX       999999     886933    Vitor   

          Cliente  
1862   LogiMaster  
1863  Velocitrans  
1864    RápidoLog  
1865    MaxiCarga  
1411  RápidoFrete

Olhando o número de valores únicos de 'ID', 'NotaEntrada' e 'NotaSaida', é possível dizer que não existem dados repetidos na tabela de vendas.

Por outro lado ao olhar a distribuição de valores na coluna 'NotaEntrada', existe uma nota de número 0 e uma de número 999999, que saem do padrão dos outros valores. Mas também é possível observar que os valores dessa coluna seguem uma incrementação padrão de 1 em 1 a partir do primeiro valor 448855. Com isso em mente talvez eu consiga utilizar da coluna 'ID' para identificar qual seria o número correto dessas notas.

In [14]:
# procurando o valores por 'ID' das linhas com a 'NotaEntrada' 0 e 999999. Procurei por um anterior, o valor incorreto e o próximo,
# pra ter certeza que a lógica de incremento está correta
vendas[vendas['ID'].isin([955,956,957]) | vendas['ID'].isin([1411,1412,1413])]

ID  DataVenda ModeloComercial  NotaEntrada  NotaSaida Vendedor  \
954    955 2023-12-06             FMX       449809     886476    Vitor   
955    956 2024-01-12              VM            0     886477    Andre   
956    957 2023-02-14              FH       449811     886478  Jessica   
1410  1411 2023-02-07              VM       450265     886932  Jessica   
1411  1412 2023-05-22             FMX       999999     886933    Vitor   
1412  1413 2023-12-11              VM       450267     886934  Jessica   

          Cliente  
954   RápidoFrete  
955     RápidoLog  
956   SuperCargas  
1410     ViaCargo  
1411  RápidoFrete  
1412     ViaCargo

Agora que eu sei que a lógica de incrementação está correta, eu vou apenas verificar se os valores que viriam a seguir não existem em alguma outra linha e depois fazer a alteração

In [15]:
# verificando se os valores que seriam adicionados a seguir existem
vendas[vendas['NotaEntrada'].isin([449810,450266])]

Empty DataFrame
Columns: [ID, DataVenda, ModeloComercial, NotaEntrada, NotaSaida, Vendedor, Cliente]
Index: []

In [16]:
# como os valores não existem na coluna 'NotaEntrada', irei subistituir os valores que se encontram fora do padrão por eles

vendas.loc[vendas['ID']==956,'NotaEntrada'] = 449810
vendas.loc[vendas['ID']==1412,'NotaEntrada'] = 450266

# e conferir como ficou depois da alteração
vendas[vendas['ID'].isin([955,956,957]) | vendas['ID'].isin([1411,1412,1413])]

ID  DataVenda ModeloComercial  NotaEntrada  NotaSaida Vendedor  \
954    955 2023-12-06             FMX       449809     886476    Vitor   
955    956 2024-01-12              VM       449810     886477    Andre   
956    957 2023-02-14              FH       449811     886478  Jessica   
1410  1411 2023-02-07              VM       450265     886932  Jessica   
1411  1412 2023-05-22             FMX       450266     886933    Vitor   
1412  1413 2023-12-11              VM       450267     886934  Jessica   

          Cliente  
954   RápidoFrete  
955     RápidoLog  
956   SuperCargas  
1410     ViaCargo  
1411  RápidoFrete  
1412     ViaCargo

Agora parece correto

Irei agora verificar os valores das outras colunas buscando incosistências

In [17]:
# verificando as data da primeira e última venda realizada
""" como ao executar a troca de tipo da coluna de  data anteriormente não ocorreu nenhum erro, todos os dados estão 
dentro do padrão e por isso vou apenas verificar se o período entre a primeira e última data fazem sentido"""
print("Data primeira venda : ",vendas['DataVenda'].min(),", Data última venda : ",vendas['DataVenda'].max())

Data primeira venda :  2023-01-02 00:00:00 , Data última venda :  2024-07-01 00:00:00


In [18]:
# verificando quais foram os modelos comerciais registrados e a distribuição deles
vendas['ModeloComercial'].value_counts()

VM     630
FMX    618
FH     617
FHX      1
Name: ModeloComercial, dtype: int64

Hoje no site da Volvo, apenas são mostrados os modelos FH,FM,FMX e VM. O modelo FHX não aparece no site da empresa, e não aparece em nenhuma busca registro de tal modelo.

Tendo isso em vista, possívelmente o modelo FHX foi inserido de forma errada, mas como tanto FH como FMX estão próximos em gramática do que foi inserido, não posso supor nenhum dos dois apenas pela gramática, para tentar encontrar o modelo eu poderia usar os dados de valores das outras tabelas para tentar achar padroes nos modelos e encontrar a qual modelo essa venda pertence, mas com 1866 vendas, fazer isso por apenas um registro, não acredito que valha a pena, por isso para o intuito de analisar os dados irei apenas excluir o registro e seus valores associados

In [19]:
# buscando a linha com o modelo FHX para saber o index para a exclusão
vendas[vendas['ModeloComercial']=='FHX']

ID  DataVenda ModeloComercial  NotaEntrada  NotaSaida Vendedor  \
25  26 2023-07-31             FHX       448880     885547    Vitor   

          Cliente  
25  EficientTrans

In [20]:
# excluindo do dataset
vendas = vendas.drop(25)

In [21]:
# verificando os modelos novamente após a exclusão
vendas['ModeloComercial'].value_counts()

VM     630
FMX    618
FH     617
Name: ModeloComercial, dtype: int64

In [22]:
# verificando quais foram os vendedores registrados e a distribuição de vendas deles
vendas['Vendedor'].value_counts()

Vitor      575
Jessica    544
Rafael     401
Andre      345
Name: Vendedor, dtype: int64

In [23]:
# verificando quais foram os clientes registrados e a distribuição de compras deles
vendas['Cliente'].value_counts()

LogiMaster       231
RápidoFrete      208
MoviFrete        197
Velocitrans      191
EficientTrans    176
SuperCargas      151
RápidoLog        114
MaxiCarga        113
FastFrete        112
RotaCerta        104
GlobalLog        103
ViaCargo          84
ExpressoNet       81
Name: Cliente, dtype: int64

As outras colunas de vendas não aparentam ter nenhuma anormalidade.

Passarei agora para os dados da tabela 'custo'

## Custo

In [24]:
# conferindo as informações de cada coluna da tabela 'custo'

custo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1866 entries, 0 to 1865
Data columns (total 8 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Nota           1866 non-null   int64         
 1   DataEmissao    1866 non-null   datetime64[ns]
 2   ValorFaturado  1866 non-null   float64       
 3   Unnamed: 3     0 non-null      float64       
 4   Unnamed: 4     0 non-null      float64       
 5   Unnamed: 5     0 non-null      float64       
 6   Unnamed: 6     0 non-null      float64       
 7   Unnamed: 7     0 non-null      float64       
dtypes: datetime64[ns](1), float64(6), int64(1)
memory usage: 116.8 KB


In [25]:
# demonstração dos dados de custos
custo.head()

Nota DataEmissao  ValorFaturado  Unnamed: 3  Unnamed: 4  Unnamed: 5  \
0  448855  2023-12-07  618653.463498         NaN         NaN         NaN   
1  448856  2024-01-25  567373.291714         NaN         NaN         NaN   
2  448857  2023-12-11  212692.573325         NaN         NaN         NaN   
3  448858  2024-01-15  429889.770092         NaN         NaN         NaN   
4  448859  2023-12-21  426597.241984         NaN         NaN         NaN   

   Unnamed: 6  Unnamed: 7  
0         NaN         NaN  
1         NaN         NaN  
2         NaN         NaN  
3         NaN         NaN  
4         NaN         NaN

Os tipos das colunas já estão de acordo com o esperado, mas foram puxadas 5 colunas em branco das planilhas. Antes de analisar os dados de cada coluna irei excluir essas colunas em branco

In [26]:
# selecionando apenas as colunas que contém dados
custo = custo[['Nota','DataEmissao','ValorFaturado']]
custo.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1866 entries, 0 to 1865
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Nota           1866 non-null   int64         
 1   DataEmissao    1866 non-null   datetime64[ns]
 2   ValorFaturado  1866 non-null   float64       
dtypes: datetime64[ns](1), float64(1), int64(1)
memory usage: 43.9 KB


In [27]:
# olhando as medidas de resumo das colunas de tipo numérico
custo.describe()

Nota  ValorFaturado
count    1866.000000    1866.000000
mean   449787.500000  433188.199830
std       538.812119  145755.473623
min    448855.000000  185096.374886
25%    449321.250000  268182.027783
50%    449787.500000  486526.815436
75%    450253.750000  551657.098073
max    450720.000000  702439.614667

In [28]:
# verificando a quantidade de valores únicos de cada coluna
custo.nunique()

Nota             1866
DataEmissao       543
ValorFaturado    1866
dtype: int64

In [29]:
# verificando as data da primeira e última venda realizada
print("Data primeira venda : ",custo['DataEmissao'].min(),", Data última venda : ",custo['DataEmissao'].max())

Data primeira venda :  2022-12-03 00:00:00 , Data última venda :  2024-06-25 00:00:00


Os dados da tabela 'custo' não apresentam nenhuma anomalia.

Passarei agora para os dados da tabela saída

## Saída

In [30]:
# conferindo as informações de cada coluna da tabela 'saida'
saida.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1866 entries, 0 to 1865
Data columns (total 3 columns):
 #   Column         Non-Null Count  Dtype         
---  ------         --------------  -----         
 0   Nota           1866 non-null   object        
 1   DataEmissao    1866 non-null   datetime64[ns]
 2   ValorFaturado  1866 non-null   float64       
dtypes: datetime64[ns](1), float64(1), object(1)
memory usage: 43.9+ KB


In [31]:
# demonstração dos dados de saída
saida.head()

Nota DataEmissao  ValorFaturado
0  885522-a prazo  2024-01-06  648937.199473
1  885523-a vista  2024-02-14  588968.814928
2  885524-a prazo  2024-01-05  220788.138400
3  885525-a prazo  2024-02-10  484838.086570
4  885526-a prazo  2024-01-12  441307.491708

In [32]:
# olhando medidas de resumo das colunas de tipo numérico
saida.describe()

ValorFaturado
count    1866.000000
mean   469606.154739
std    157361.558513
min       115.000000
25%    290703.342627
50%    530272.789666
75%    601282.030656
max    714345.370847

In [33]:
# verificando a quantidade de valores únicos de cada coluna
saida.nunique()

Nota             1866
DataEmissao       532
ValorFaturado    1866
dtype: int64

In [34]:
# verificando as data da primeira e última venda realizada
print("Data primeira venda : ",saida['DataEmissao'].min(),", Data última venda : ",saida['DataEmissao'].max())

Data primeira venda :  2023-01-04 00:00:00 , Data última venda :  2024-07-08 00:00:00


Os tipos das colunas já estão de acordo com o esperado, e não aparecereu nenhuma anomalia nos dados. Porém antes de prosseguir irei dividir a coluna 'Nota' em duas, para que possa ter uma coluna apenas com o número da nota e uma com o tipo de pagamento,e  passar o tipo de dado da coluna 'Nota' para int

In [35]:
# separando a antiga coluna 'Nota' e colocando a parte de tipo de pagamento em outra coluna

nota_tipo = saida['Nota'].str.split('-')
saida['Nota'],saida['Tipo pagamento'] = nota_tipo.str.get(0),nota_tipo.str.get(1)

In [36]:
# verificando como ficaram as colunas
saida

Nota DataEmissao  ValorFaturado Tipo pagamento
0     885522  2024-01-06  648937.199473        a prazo
1     885523  2024-02-14  588968.814928        a vista
2     885524  2024-01-05  220788.138400        a prazo
3     885525  2024-02-10  484838.086570        a prazo
4     885526  2024-01-12  441307.491708        a prazo
...      ...         ...            ...            ...
1861  887383  2024-02-29  559633.313251        a prazo
1862  887384  2024-06-22  271356.489909        a prazo
1863  887385  2023-06-22  538306.153495        a prazo
1864  887386  2023-12-31  524499.459237        a prazo
1865  887387  2024-02-19  674615.832576        a prazo

[1866 rows x 4 columns]

In [37]:
# trocando o tipo da coluna 'Nota' e conferindo o resultado
saida['Nota'] = saida['Nota'].astype(int)
saida.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1866 entries, 0 to 1865
Data columns (total 4 columns):
 #   Column          Non-Null Count  Dtype         
---  ------          --------------  -----         
 0   Nota            1866 non-null   int32         
 1   DataEmissao     1866 non-null   datetime64[ns]
 2   ValorFaturado   1866 non-null   float64       
 3   Tipo pagamento  1866 non-null   object        
dtypes: datetime64[ns](1), float64(1), int32(1), object(1)
memory usage: 51.1+ KB


In [38]:
# verificando a coluna 'Tipo pagamento' pra ter certeza que não aconteceu nenhum erro na criação dela
saida['Tipo pagamento'].value_counts()

a prazo    1381
a vista     485
Name: Tipo pagamento, dtype: int64

Com isso a tabela 'saida' fica concluída e resta agora apenas a tabela caixa

## Caixa

In [39]:
# conferindo as informações de cada coluna da tabela 'caixa'
caixa.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14669 entries, 0 to 14668
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Nota         14669 non-null  int64  
 1   Parcela      14669 non-null  int64  
 2   Valor        14669 non-null  float64
 3   DataEntrada  14669 non-null  object 
dtypes: float64(1), int64(2), object(1)
memory usage: 458.5+ KB


In [40]:
# demonstração dos dados de caixa
caixa.head()

Nota  Parcela          Valor DataEntrada
0  885523        0  588968.814928   2024-2-14
1  885528        0  525959.972055    2024-1-7
2  885532        0  227605.407167   2024-1-17
3  885533        0  486270.971853   2024-2-16
4  885534        0  588344.545925   2024-1-11

Nenhum valor nulo encontrado e nenhuma anomalia aparente, porém a coluna 'DataEntrada' precisa ser formatada como data

In [41]:
# mudando o tipo de dado da coluna 'DataEntrada' e conferindo o resultado
caixa['DataEntrada'] = pd.to_datetime(caixa['DataEntrada'])
caixa.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 14669 entries, 0 to 14668
Data columns (total 4 columns):
 #   Column       Non-Null Count  Dtype         
---  ------       --------------  -----         
 0   Nota         14669 non-null  int64         
 1   Parcela      14669 non-null  int64         
 2   Valor        14669 non-null  float64       
 3   DataEntrada  14669 non-null  datetime64[ns]
dtypes: datetime64[ns](1), float64(1), int64(2)
memory usage: 458.5 KB


In [44]:
# verificando as data do primeiro e último registro em caixa realizado
print("Data primeira venda : ",saida['DataEmissao'].min(),", Data última venda : ",saida['DataEmissao'].max())

Data primeira venda :  2023-01-04 00:00:00 , Data última venda :  2024-07-08 00:00:00


In [42]:
# olhando medidas de resumo das colunas de tipo numérico
caixa.describe()

Nota       Parcela          Valor
count   14669.000000  14669.000000   14669.000000
mean   886441.166269      5.776058   20273.868681
std       534.836659      4.590984   88171.629781
min    885522.000000      0.000000    1537.526238
25%    885976.000000      2.000000    3449.626256
50%    886440.000000      5.000000    4562.697767
75%    886889.000000      9.000000    6345.716881
max    887387.000000     18.000000  714345.370847

In [43]:
# verificando a quantidade de valores únicos de cada coluna
caixa.nunique()

Nota           1866
Parcela          19
Valor          2528
DataEntrada     545
dtype: int64

Finalizada a limpeza na tabela 'caixa', também finalizam as limpezas de forma geral e agora que todas as tabelas estão com os dados em conformidade, irei terminar de extrair os dados do registro que teve o modelo inserido de forma incorreta na tabela 'vendas'.

Como verificado antes da exclusão registro na tabela de vendas, as notas correspondentes à esse registro são 448880 nota de entrada, e 885547 nota de saída.

In [45]:
# excluindo os registros dessa venda na tabela 'custo'

custo_index = custo[custo['Nota']==448880].index.values
custo = custo.drop(custo_index)

# excluindo os registros dessa venda na tabela 'saida'

saida_index = saida[saida['Nota']==885547].index.values
saida = saida.drop(saida_index)

# excluindo os registros dessa venda na tabela custo

caixa_index = caixa[caixa['Nota']==885547].index.values
caixa = caixa.drop(caixa_index)


## Exportando os dados

In [46]:
# exportando os dados atualizados para arquivos csv que serão usados para a análise
vendas.to_csv('vendas.csv',encoding='utf-8',header=True,index=False)
custo.to_csv('custo.csv',encoding='utf-8',header=True,index=False)
saida.to_csv('saida.csv',encoding='utf-8',header=True,index=False)
caixa.to_csv('caixa.csv',encoding='utf-8',header=True,index=False)

In [ ]:
from powerbiclient import Report, models

# Import the DeviceCodeLoginAuthentication class to authenticate against Power BI
from powerbiclient.authentication import DeviceCodeLoginAuthentication

# Initiate device authentication
device_auth = DeviceCodeLoginAuthentication()

group_id="Workspace ID"
report_id="Report ID"

report = Report(group_id=group_id, report_id=report_id, auth=device_auth)

report

In [1]:
<iframe title="Análise Treviso 2.0" width="600" height="373.5" src="https://app.powerbi.com/view?r=eyJrIjoiMmExYzVhZmQtYmJhMS00YTFkLTkzNzMtOTk0YTE2ZWMwZGJiIiwidCI6IjZjY2M4NmNjLTY2YjItNGEwMS04YzZmLWUyMzAwODA4MTc4NyJ9" frameborder="0" allowFullScreen="true"></iframe>

SyntaxError: invalid syntax (<ipython-input-1-abd2dd4caf49>, line 1)